engine/execution/ingestion/engine.go

```diff
@@ -210,7 +210,7 @@ func (e *Engine) BlockProcessable(b *flow.Header, _ *flow.QuorumCertificate) {
 		return
 	}
 
-	blockID := b.ID()
+	blockID := qc.BlockID
```

Comment:
The caller guarantees the qc is for the block ID, this allows us to save a ID() call